### DISCUSSION TITLE
Explanation of --trace output

### DISCUSSION THREAD

**Discussion by DrBanana419** [netblue30/firejail#5427] [Q&A]
I looked online but couldn't find anything. So, what do these mean:

```
open /path/to/file
access /path/to/file
open64 /path/to/file
stat /path/to/file
fopen /path/to/file
unlink /path/to/file
connect /path/to/file
socket /path/to/file
bind /path/to/file
```

Also, what do the trailing 0 and -1 mean? I'm assuming 0 means it opened succesfully, and -1 means it failed?

---

EDIT by @rusty-snake: fix code-block

**Comment by rusty-snake** [comment]
Actually every call has it's own format. However there are common pattern like for filesystem related calls:

```
<pid>:<comm>:<call> <pathname>:<rv>
```
- pid: The pid of the program. This is the pid inside the sandbox.
- comm: The name of the command (`/proc/<pid>/comm`).
- call: The name of the called function.
- pathname: The pathname to operate on (e.g. open, stat, ...)
- rv: The return value of the called libc function. And yes `-1` means error.

Or setuid related calls:

```
<pid>:<comm>:<call> <uid> <uid2> <...>:<rv>
```
